drv/sidecar-mainboard-controller/src/tofino2.rs

```diff
@@ -231,3 +232,435 @@ impl Sequencer {
         })
     }
 }
+
+bitfield! {
+    #[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+    #[repr(C)]
+    pub struct DebugPortState(u8);
+    pub send_buffer_empty, set_send_buffer_empty: 0;
+    pub send_buffer_full, _: 1;
+    pub receive_buffer_empty, set_receive_buffer_empty: 2;
+    pub receive_buffer_full, _: 3;
+    pub request_in_progress, set_request_in_progress: 4;
+    pub address_nack_error, set_address_nack_error: 5;
+    pub byte_nack_error, set_byte_nack_error: 6;
+}
+
+#[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+#[repr(u8)]
+pub enum DebugRequestOpcode {
+    LocalWrite = 0,
+    LocalRead = 32,
+    DirectWrite = 128,
+    DirectRead = 160,
+    IndirectWrite = 192,
+    IndirectRead = 224,
+}
+
+impl From<DebugRequestOpcode> for u8 {
+    fn from(opcode: DebugRequestOpcode) -> Self {
+        opcode as u8
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+#[repr(u8)]
+pub enum DirectBarSegment {
+    Bar0 = 0,
+    Msi = 1,
+    Cfg = 2,
+}
+
+impl From<DirectBarSegment> for u32 {
+    fn from(segment: DirectBarSegment) -> Self {
+        match segment {
+            DirectBarSegment::Bar0 => 0,
+            DirectBarSegment::Msi => (1 as u32) << 28,
+            DirectBarSegment::Cfg => (2 as u32) << 28,
+        }
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+#[repr(u32)]
+pub enum TofinoRegisters {
+    Scratchpad = 0x0,
+    FreeRunningCounter = 0x10,
+    SpiOutData0 = (0x80000 | 0x120),
+    SpiOutData1 = (0x80000 | 0x124),
+    SpiInData = (0x80000 | 0x12c),
+    SpiCommand = (0x80000 | 0x128),
+    SpiIdCode = (0x80000 | 0x130),
+}
+
+impl From<TofinoRegisters> for u32 {
+    fn from(r: TofinoRegisters) -> Self {
+        r as u32
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+#[repr(u8)]
+pub enum SpiEepromInstruction {
```

Comment:
Add a comment explaining that this comes from the EEPROM, and maybe add the command acronyms (`WREN`, etc) in comments?